nestkernel/conn_parameter.h

```diff
@@ -82,6 +82,13 @@ class ConnParameter
   }
   virtual bool is_array() const = 0;
 
+  virtual void
+  reset() const
+  {
+    throw NotImplemented(
+      "Symmetric connections do not work with random parameters." );
```

Comment:
All kinds of future parameter types might not support resetting. "Symmetric connections require parameters that can be reset." would be more general.